### ISSUE TITLE
Optimize is part of the unified release process

### ISSUE THREAD

**Issue by buccarel** [camunda/camunda#22351]
Engineering DRI: @grlimacan

### Context

Once Optimize is part of the single application, it shall be released as a single JAR together with the rest of the stack.

Blocked by https://github.com/camunda/camunda/issues/22271

Key goals:
- Include Optimize in the technical release build (maven build and/or GitHub workflow)
- Include Optimize in webapp QA

### Acceptance Criteria

- [x] GHA: A sub workflow is added in the [platform release workflow](https://github.com/camunda/camunda/blob/main/.github/workflows/camunda-platform-release.yml) to push the optimize docker image. (Here is example of the [operate version](https://github.com/camunda/camunda/blob/main/.github/workflows/camunda-platform-release.yml#L404))
- [ ] GHA/pom: ~~The Optimize release is integrated in the common release workflow by removing the skip from the maven command of the [platform release workflow](https://github.com/camunda/camunda/blob/main/.github/workflows/camunda-platform-release.yml). Other skips may also need to be released~~ Solved with #22271 
- [x] GHA: Existing Optimize release workflows that are not part of the unified release process are removed

### Links

* Part of https://github.com/camunda/product-hub/issues/2289
* [conversation on how to make Optimize part of unified release](https://camunda.slack.com/archives/C03NFMH4KC6/p1727775547558379)
* [other conversation on what is the main workflow file](https://camunda.slack.com/archives/C06UWQNCU7M/p1729494968757159)
* [monorepo release process repo](https://github.com/zeebe-io/zeebe-engineering-processes/tree/main/src/main/resources/release)
* https://github.com/camunda/Release-Train/issues/21

### Breakdown

```[tasklist]
### Sub tasks
- [ ] https://github.com/camunda/camunda/issues/25174
```

```[tasklist]
### Pull requests
- [ ] https://github.com/camunda/camunda/pull/26081
- [ ] https://github.com/zeebe-io/zeebe-engineering-processes/pull/464
```

## Task Lifecycle

For managing the issue lifecycle, please use the workflow commands. You can see the available
commands by writing `/help` as a comment on this issue.

#### Engineering Review

- [ ] All code targeting the main branch has been reviewed by at least one Engineer
- [ ] The PR targeting the main branch has been approved by the reviewing Engineer
- [ ] If the API has changed, the API documentation is updated
- [ ] All other PRs (docs, controller etc.) in the breakdown have been approved

#### QA Review

- [ ] The change is implemented as described on all target environments/versions/modes
- [ ] The documentation changes are as expected

### Completion

- [ ] All Review stages are successfully completed
- [ ] All associated PRs are merged to the main branch(es) and stable branches
- [ ] The correct version labels are applied to the issue

**Comment by misiekhardcore** [comment]
the issue description says: **Existing Optimize release workflows that are not part of the unified release process are removed**, but I think we are actually keeping them to perform patches?

**Comment by RomanJRW** [comment]
@misiekhardcore - I think this issue description was written before we knew that patches would have to be handled on a dedicated Optimize branch. Now this is known, I think your comment is true that we would need to keep them

**Comment by misiekhardcore** [comment]
@RomanJRW I am unassigning myself from this one due to the team switch.

**Comment by OmranAbazid** [comment]
These are the high level steps that we need to do to complete this:
- Optimize is part of the single Jar
- Optimize is using the common "dist" module to build optimize instead of using the "optimize-distro"
- Remove the skip from the maven command of the [platform release workflow](https://github.com/camunda/camunda/blob/main/.github/workflows/camunda-platform-release.yml)
- Add a sub workflow in the [platform release workflow](https://github.com/camunda/camunda/blob/main/.github/workflows/camunda-platform-release.yml) to push the optimize docker image. (Here is example of the [operate version](https://github.com/camunda/camunda/blob/main/.github/workflows/camunda-platform-release.yml#L404))
- Integrate the Optimize release process into the camunda release process.

**Comment by grlimacan** [comment]
@RomanJRW Here are some open questions whether the scope of this Issue includes the following:

1. Make all critical Optimize CI checks part of the merge queue checks in the mono repo
2. Include Optimize in webapp QA
3. Integrate the Optimize release process model into the camunda release process.: As I understand there are no product specific steps in the mono-repo release process. Or if you know any such places, can you please link them here? If my assumption is correct, as long as Optimize is technically part of the GH release workflow, are there still changes that we need to do there?
4. No changes will be done to the stable branches, this concerns exclusively our `main` repository with the 8.7.0 release in mind, correct?

**Comment by RomanJRW** [comment]
**From discussion:**
1. Unit tests already there. Upgrade tests would be a good addition if they match the required criteria, Removing from scope here because it's already achieved
2. Find out more on Friday with Schoetti
3. Find out more on Friday with Schoetti
4. Yes, no intentional change to 8.6. This should stay the same. We do not need to move to the shared stable branch

**Comment by grlimacan** [comment]
both PRs are merged. Only the sub-task is still open, so closing the issue
